File: java/src/main/java/dev/medzik/libcrypto/Random.java
package dev.medzik.libcrypto;

import java.security.SecureRandom;

public final class Random {
    /** Generate a random byte array. */
    public static byte[] randBytes(int length) {
        SecureRandom rd = new SecureRandom();
        byte[] salt = new byte[length];
        rd.nextBytes(salt);
        return salt;
    }
}

File: javascript/src/random.ts
import { Buffer } from "buffer/"

/**
 * Generate a random byte array of the given length
 * @param length The length of the salt to generate
 */
export function randBytes(length: number): Buffer {
    const salt: number[] = []
    for (let i = 0; i < length; i++) {
        salt.push(Math.floor(Math.random() * 255))
    }

    return Buffer.from(salt)
}
